src/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs

```diff
@@ -290,11 +291,42 @@ public Blend Blend
             }
             set
             {
-                // Allocate temporary native memory buffer
-                // and copy input blend factors into it.
+                if (value == null || value.Factors == null)
+                {
+                    // This is the behavior on Desktop
+                    throw new NullReferenceException();
+                }
+
+                if (value.Positions == null)
+                {
+                    throw new ArgumentNullException("source");
```

Comment:
It's not right, but it's the behavior on Desktop and there are unit tests asserting this.

The Desktop throws  `ArgumentNullException("source")` because it never validates the value of `value.Positions`, and then passes it on to `Marshal.Copy(value.Positions, 0, positions, count);`. The first argument of `Marshal.Copy` is `source`, hence this exception.

I'd be more than happy to change it into a more sensible error message, if you're OK with changing the exception contract (which arguably is broken on Desktop).